StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1203ConstantsMustAppearBeforeFields.cs

```diff
@@ -42,7 +45,53 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleTypeDeclaration, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleTypeDeclaration, SyntaxKind.StructDeclaration);
+        }
+
+        private static void HandleTypeDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var typeDeclaration = context.Node as TypeDeclarationSyntax;
+
+            if (typeDeclaration == null)
+            {
+                return;
+            }
+
+            var members = typeDeclaration.Members;
+            bool nonConstFieldFound = false;
+
+            for (int i = 0; i < members.Count; i++)
+            {
+                if (members[i].IsKind(SyntaxKind.IncompleteMember))
```

Comment:
:bulb: This check and the one following it are redundant. The only thing you need is this:

```csharp
var field = members[i] as FieldDeclarationSyntax;
if (field == null)
{
  continue;
}
```